### DISCUSSION TITLE
AlertDialog -- showing loading spinner on confirm button

### DISCUSSION THREAD

**Discussion by ahcai** [twilio-labs/paste#1795] [Q&A] [answered]
The alert dialog is often used to ask for confirmation before an action is taken and sometimes this action takes a while so we need a way to show a loading spinner on the confirm button. It would be useful if there was a `confirmIsLoading` prop to decide whether the confirm button would show a loading spinner (or if this is too customized, an alternative is allowing a complete override of the dialog footer).

**Comment by kate-porter** [comment]
Hey @ahcai - thanks for this question! We are discussing as a team and will get back to you soon.

**Reply by kate-porter** [reply]
Hey @ahcai - we chatted about this, and we've created a ticket to adjust AlertDialog to allow for this. Here's the [ticket](https://issues.corp.twilio.com/browse/DSYS-3009) if you want to watch it. Once that ticket is done, you'll be able to set the AlertDialog button to Loading (or any other action triggers that are available on Button, like Disabled, etc.).

Do consider, though, if keeping the user trapped in an AlertDialog while the action processes is a great user experience, especially if the process will take awhile. There may be a different way to indicate that the process is in progress without keeping the user stuck in the dialog (perhaps a Spinner or note inline next to the thing that is currently being deleted).

@TheSisb also made some alternative recommendations for a similar use case on [this GH Discussion](https://github.com/twilio-labs/paste/discussions/1776) that you could consider.

All that said, if you determine that the loading state on the AlertDialog Button is indeed the best user experience, you'll be able to do that soon!

Let us know if you have any other questions on this. Thanks for pointing it out!

**Reply by ahcai** [reply]
Thanks @katieporter!

**Reply by ahcai** [reply]
Do you have an estimate of when this ticket would be done by any chance?

**Reply by TheSisb** [reply]
Hi @ahcai, just wanted to report back that we don't currently have an estimate for when this will get done. We'll report back when we get around to planning the ticket. Sorry for the inconvenience.

**Comment by loreina** [comment, marked as answer]
Marking this as answered due to inactivity.
